### ISSUE TITLE
DTLS/ICE bridging + VPN: "SRTCP/SRTP output wanted, but no crypto suite was negotiated"

### ISSUE THREAD

**Issue by space88man** [sipwise/rtpengine#1114]
Problem: version mr9.0.1.5. UA with wireguard VPN address making DTLS/ICE call with `ICE=force DTLS=passive`.  The UA is PJSIP 2.10.

When the UA tries all its IP addresses(including wireguard VPN) as ICE candidates I reproducibly get the following ERR messages repeated:
1. `SRTP output wanted, but no crypto suite was negotiated` OR
1. `SRTCP output wanted, but no crypto suite was negotiated`; if this case happens there is no MOS report
at rtpengine.

when both peers are using DTLS/ICE. The call appears to connect correctly and media is exchanged between the two peers.

The UA has many interfaces — if the UA is forced to bind to the VPN address it works without this error. The NG flow is
```
INVITE: rtcp-mux-accept rtcp-mux-offer DTLS=passive ICE=force
ANSWER: ICE=force
# the PJSIP UA always sends UPDATE when using DTLS/ICE, not quite sure why,  so


UPDATE: rtcp-mux-accept rtcp-mux-offer ICE=force
ANSWER: ICE=force
```

1. No STUN/TURN: address is determined by ICE prflx. The confirmed `peer address` is always correct, and ICE negotiation manages to find the VPN address as the endpoint.
1. The call works fine if one side is using RTP/SAVP; the problem is specific to both sides using UDP/RTP/TLS/SAVP.
1. This effect is reproducible, but the complaint is either SRTP or SRTCP, never both. When rtpengine complains about SRTP, then I do get MOS stats (I guess SRTCP is working then).
1. When neither party is on VPN, the configuration works perfectly 
1. If the callee is on RTP/SAVP, I will see only once

        SRTP output wanted, but no crypto suite was negotiated
        SRTCP output wanted, but no crypto suite was negotiated
but after the call proceeds with no more error messages, and MOS is reported

How would I start to troubleshooting this weird situation?

```
# This is the port in the INVITE to callee
# but this port is changed in the UPDATE 
rtpengine[7567]: [1606050070.836458] ERR: SRTP output wanted, but no crypto suite was negotiated
rtpengine[7567]: [1606050080.022175] ERR: [Z7jmd5tTP6CtLlvnWuLZB34vx2HJDgyy port 16634]: SRTP output wanted, but no crypto suite was negotiated
rtpengine[7567]: [1606050086.560113] ERR: SRTP output wanted, but no crypto suite was negotiated
rtpengine[7567]: [1606050095.017568] ERR: [Z7jmd5tTP6CtLlvnWuLZB34vx2HJDgyy port 16634]: SRTP output wanted, but no crypto suite was negotiated
rtpengine[7567]: [1606050102.280212] ERR: SRTP output wanted, but no crypto suite was negotiated
rtpengine[7567]: [1606050110.013203] ERR: [Z7jmd5tTP6CtLlvnWuLZB34vx2HJDgyy port 16634]: SRTP output wanted, but no crypto suite was negotiated
rtpengine[7567]: [1606050117.993698] ERR: SRTP output wanted, but no crypto suite was negotiated
rtpengine[7567]: [1606050125.016974] ERR: [Z7jmd5tTP6CtLlvnWuLZB34vx2HJDgyy port 16634]: SRTP output wanted, but no crypto suite was negotiated
rtpengine[7567]: [1606050133.712417] ERR: SRTP output wanted, but no crypto suite was negotiated
rtpengine[7567]: [1606050140.012736] ERR: [Z7jmd5tTP6CtLlvnWuLZB34vx2HJDgyy port 16634]: SRTP output wanted, but no crypto suite was negotiated
rtpengine[7567]: [1606050149.438970] ERR: SRTP output wanted, but no crypto suite was negotiated
rtpengine[7567]: [1606050155.018120] ERR: [Z7jmd5tTP6CtLlvnWuLZB34vx2HJDgyy port 16634]: SRTP output wanted, but no crypto suite was negotiated
rtpengine[7567]: [1606050165.158182] ERR: SRTP output wanted, but no crypto suite was negotiated
rtpengine[7567]: [1606050170.022259] ERR: [Z7jmd5tTP6CtLlvnWuLZB34vx2HJDgyy port 16634]: SRTP output wanted, but no crypto suite was negotiated
rtpengine[7567]: [1606050180.873904] ERR: SRTP output wanted, but no crypto suite was negotiated
rtpengine[7567]: [1606050185.008373] ERR: [Z7jmd5tTP6CtLlvnWuLZB34vx2HJDgyy port 16634]: SRTP output wanted, but no crypto suite was negotiated
rtpengine[7567]: [1606050196.590326] ERR: SRTP output wanted, but no crypto suite was negotiated
rtpengine[7567]: [1606050200.013028] ERR: [Z7jmd5tTP6CtLlvnWuLZB34vx2HJDgyy port 16634]: SRTP output wanted, but no crypto suite was negotiated
rtpengine[7567]: [1606050212.316752] ERR: SRTP output wanted, but no crypto suite was negotiated
rtpengine[7567]: [1606050215.010216] ERR: [Z7jmd5tTP6CtLlvnWuLZB34vx2HJDgyy port 16634]: SRTP output wanted, but no crypto suite was negotiated
rtpengine[7567]: [1606050228.037941] ERR: SRTP output wanted, but no crypto suite was negotiated
rtpengine[7567]: [1606050230.014001] ERR: [Z7jmd5tTP6CtLlvnWuLZB34vx2HJDgyy port 16634]: SRTP output wanted, but no crypto suite was negotiated
```

**Comment by space88man** [comment]
Update with STUN

1. if I force the UA to include only 1 host candidate (the one pointing to the default route), I get the same issue, even though the VPN tunnel endpoint is discovered by ICE(prflx)
1. if I add the same host candidate and 1 srflx candidate from a STUN server(the VPN endpoint), it works correctly.

**Comment by space88man** [comment]
SIP from caller, single host candidate that reproduces the problem

```
INVITE sip:alice@example.com SIP/2.0
Via: SIP/2.0/TLS 10.13.20.1:36806;rport;branch=z9hG4bKPjMTkXxam6y3LZEXqHm5aKEUMx4I.7NmXY;alias
Max-Forwards: 70
From: sip:mary@example.com;tag=aNGUz7DpGyYBReacmrr-2EqYKtf9F-dR
To: sip:alice@example.com
Contact: <sip:mary@10.13.20.1:36806;transport=TLS;extension=webrtc;ob>;+sip.ice
Call-ID: Ot1VXA-3WM-ta-1LZKiE2IPS-8682voP
CSeq: 4216 INVITE
Route: <sip:example.com:5063;transport=tls;lr>
Allow: PRACK, INVITE, ACK, BYE, CANCEL, UPDATE, INFO, SUBSCRIBE, NOTIFY, REFER, MESSAGE, OPTIONS
Supported: replaces, 100rel, timer, norefersub
Session-Expires: 1800
Min-SE: 90
User-Agent: PJSUA v2.10-dev Linux-5.8.19.44/x86_64/glibc-2.32
Proxy-Authorization: ...
Content-Type: application/sdp
Content-Length:   734

v=0
o=- 3815032388 3815032388 IN IP4 192.168.1.7
s=pjmedia
b=AS:117
t=0 0
a=X-nat:0
m=audio 4024 UDP/TLS/RTP/SAVP 96 120
c=IN IP4 192.168.1.7
b=TIAS:96000
a=rtcp:4006 IN IP4 192.168.1.7
a=sendrecv
a=rtpmap:96 opus/48000/2
a=fmtp:96 useinbandfec=1
a=rtpmap:120 telephone-event/48000
a=fmtp:120 0-16
a=ssrc:733990370 cname:4f3d15f275434345
a=ice-ufrag:0c3f8df0
a=ice-pwd:1dc4a2a94e91252c47a9a1b2
a=candidate:Hc0a80107 1 UDP 2130706431 192.168.1.7 4024 typ host
a=candidate:Hc0a80107 2 UDP 2130706430 192.168.1.7 4006 typ host
a=rtcp-mux
a=setup:actpass
a=fingerprint:SHA-256 BE:A7:35:D8:17:C9:7A:67:43:26:4B:F4:40:50:60:17:18:56:BA:ED:21:B1:75:F9:8D:DC:0C:7C:86:03:A1:B4
m=video 0 RTP/AVP 31
c=IN IP4 127.0.0.1

SIP/2.0 200 OK
Via: SIP/2.0/TLS 10.13.20.1:36806;rport=36806;received=10.13.20.1;branch=z9hG4bKPjMTkXxam6y3LZEXqHm5aKEUMx4I.7NmXY;alias
Record-Route: <sip:example.com:5063;transport=tls;lr;r2=on>
Call-ID: Ot1VXA-3WM-ta-1LZKiE2IPS-8682voP
From: <sip:mary@example.com>;tag=aNGUz7DpGyYBReacmrr-2EqYKtf9F-dR
To: <sip:alice@example.com>;tag=DLXjjtb4BTJonPCeD.gQZmOVYJW0vaAA
CSeq: 4216 INVITE
Contact: <sip:alice@10.13.20.1:47762;transport=TLS;ob>;+sip.ice
Allow: PRACK, INVITE, ACK, BYE, CANCEL, UPDATE, INFO, SUBSCRIBE, NOTIFY, REFER, MESSAGE, OPTIONS
Supported: replaces, 100rel, timer, norefersub
Session-Expires: 1800;refresher=uac
Require: timer
Content-Type: application/sdp
Content-Length:   640

v=0
o=- 3815032388 3815032389 IN IP4 192.168.12.42
s=pjmedia
b=AS:117
t=0 0
a=X-nat:1
m=audio 17894 UDP/TLS/RTP/SAVP 96 120
c=IN IP4 10.18.0.12
b=TIAS:96000
b=RS:0
b=RR:0
a=ssrc:2098581968 cname:2048942936049244
a=rtpmap:96 opus/48000/2
a=rtpmap:120 telephone-event/48000
a=fmtp:96 useinbandfec=1
a=fmtp:120 0-16
a=sendrecv
a=rtcp:17894
a=rtcp-mux
a=setup:active
a=fingerprint:sha-1 32:AB:14:A0:A0:44:54:D3:12:F7:44:B6:57:48:46:CB:00:71:69:95
a=ice-ufrag:15rYudV1
a=ice-pwd:8FJGYJ0MYa5vu76b0sOxxiZ974
a=candidate:BE2FdFiRyWOsmhMl 1 UDP 2130706431 10.18.0.12 17894 typ host
m=video 0 RTP/AVP 31
c=IN IP4 10.18.0.12

UPDATE sip:alice@10.13.20.1:47762;transport=TLS;ob SIP/2.0
Via: SIP/2.0/TLS 10.13.20.1:36806;rport;branch=z9hG4bKPjOQKGpWiE5W.fGSyvMoYEobvkeTguiVVo;alias
Max-Forwards: 70
From: sip:mary@example.com;tag=aNGUz7DpGyYBReacmrr-2EqYKtf9F-dR
To: sip:alice@example.com;tag=DLXjjtb4BTJonPCeD.gQZmOVYJW0vaAA
Contact: <sip:mary@10.13.20.1:36806;transport=TLS;extension=webrtc;ob>;+sip.ice
Call-ID: Ot1VXA-3WM-ta-1LZKiE2IPS-8682voP
CSeq: 4217 UPDATE
Route: <sip:example.com:5063;transport=tls;lr;r2=on>
Supported: replaces, 100rel, timer, norefersub
Session-Expires: 1800;refresher=uac
Min-SE: 90
Content-Type: application/sdp
Content-Length:   736

v=0
o=- 3815032388 3815032389 IN IP4 192.168.1.7
s=pjmedia
b=AS:117
t=0 0
a=X-nat:0
m=audio 4024 UDP/TLS/RTP/SAVP 96 120 
c=IN IP4 192.168.81.5
b=TIAS:96000
b=RS:0
b=RR:0
a=rtpmap:96 opus/48000/2
a=fmtp:96 useinbandfec=1
a=rtpmap:120 telephone-event/48000
a=fmtp:120 0-16
a=ssrc:733990370 cname:4f3d15f275434345
a=ice-ufrag:0c3f8df0
a=ice-pwd:1dc4a2a94e91252c47a9a1b2
a=candidate:Pc0a80107 1 UDP 1862270463 192.168.81.5 4024 typ prflx raddr 192.168.1.7 rport 4024
a=remote-candidates:1 10.18.0.12 17894
a=rtcp-mux
a=setup:passive
a=fingerprint:SHA-256 BE:A7:35:D8:17:C9:7A:67:43:26:4B:F4:40:50:60:17:18:56:BA:ED:21:B1:75:F9:8D:DC:0C:7C:86:03:A1:B4
a=sendrecv
m=video 0 RTP/AVP 31
c=IN IP4 127.0.0.1
a=inactive

SIP/2.0 200 OK
Via: SIP/2.0/TLS 10.13.20.1:36806;rport=36806;received=10.13.20.1;branch=z9hG4bKPjOQKGpWiE5W.fGSyvMoYEobvkeTguiVVo;alias
Call-ID: Ot1VXA-3WM-ta-1LZKiE2IPS-8682voP
From: <sip:mary@example.com>;tag=aNGUz7DpGyYBReacmrr-2EqYKtf9F-dR
To: <sip:alice@example.com>;tag=DLXjjtb4BTJonPCeD.gQZmOVYJW0vaAA
CSeq: 4217 UPDATE
Session-Expires: 1800;refresher=uac
Require: timer
Contact: <sip:alice@10.13.20.1:47762;transport=TLS;ob>;+sip.ice
Allow: PRACK, INVITE, ACK, BYE, CANCEL, UPDATE, INFO, SUBSCRIBE, NOTIFY, REFER, MESSAGE, OPTIONS
Supported: replaces, 100rel, timer, norefersub
Content-Type: application/sdp
Content-Length:   640

v=0
o=- 3815032388 3815032390 IN IP4 192.168.12.42
s=pjmedia
b=AS:117
t=0 0
a=X-nat:1
m=audio 17894 UDP/TLS/RTP/SAVP 96 120
c=IN IP4 10.18.0.12
b=TIAS:96000
b=RS:0
b=RR:0
a=ssrc:2098581968 cname:2048942936049244
a=rtpmap:96 opus/48000/2
a=rtpmap:120 telephone-event/48000
a=fmtp:96 useinbandfec=1
a=fmtp:120 0-16
a=sendrecv
a=rtcp:17894
a=rtcp-mux
a=setup:active
a=fingerprint:sha-1 32:AB:14:A0:A0:44:54:D3:12:F7:44:B6:57:48:46:CB:00:71:69:95
a=ice-ufrag:15rYudV1
a=ice-pwd:8FJGYJ0MYa5vu76b0sOxxiZ974 
a=candidate:BE2FdFiRyWOsmhMl 1 UDP 2130706431 10.18.0.12 17894 typ host
m=video 0 RTP/AVP 31
c=IN IP4 10.18.0.12
```

**Comment by space88man** [comment]
Single host, single srflx candidate using STUN; this combination works.

```
INVITE sip:alice@example.com SIP/2.0                                                                                           
Via: SIP/2.0/TLS 10.13.20.1:40166;rport;branch=z9hG4bKPj7X09kXL17bMarp1bKvl4pHOeG.R6fTiB;alias                                                    
Max-Forwards: 70                                                         
From: sip:mary@example.com;tag=FyIJkATEQ0xXwh-yOLw7r63M0rvemI6V                                                                
To: sip:alice@example.com                                                                                                      
Contact: <sip:mary@10.13.20.1:40166;transport=TLS;extension=webrtc;ob>;+sip.ice                                                        
Call-ID: NUTfIMsRvuEVd0mhxrXqe9AwqylV753A                                                                                                         
CSeq: 13483 INVITE                                                                                                                                
Route: <sip:example.com:5063;transport=tls;lr>                   
Allow: PRACK, INVITE, ACK, BYE, CANCEL, UPDATE, INFO, SUBSCRIBE, NOTIFY, REFER, MESSAGE, OPTIONS                                                  
Supported: replaces, 100rel, timer, norefersub                           
Session-Expires: 1800                                                                                                                             
Min-SE: 90                                                               
User-Agent: PJSUA v2.10-dev Linux-5.8.19.44/x86_64/glibc-2.32                                                                                     
Proxy-Authorization: Digest username="mary@example.com", realm="example.com", nonce="X7pKWl+6SS5T3YtmCXsnwKOTrIj2D1vf",
 uri="sip:alice@example.com", response="eyJzaWduYXR1cmUiOiJNRVVDSVFEclhQZms2d25HL3RMd0VQdDgyaStPalhmSmVESTc5VFFsV0p1UktHanFRQUl
nZEpremZLY2FwMzFmeDc0TzkrYTJ5SUZqYjhuK0dpUHNlWHFwY21DTTRUdz0iLCJhc3NlcnRpb24iOnsiaXNzdWVyX2ppZCI6InMudGJzX3Nzb0B2NmhhLmRldi5vbnRhbGsuY29tIiwic3Via
mVjdF9qaWQiOiJ1LmhhYm90cy5oYW1hcnlAdjZoYS5kZXYub250YWxrLmNvbSIsImV4cGlyZXNfYXRfdGltZXN0YW1wIjoxNjA2MDQ5NDY2LCJjcmVhdGVkX2F0IjoxNjA2MDQyMjY2fX0="  
Content-Type: application/sdp                                            
Content-Length:   932   
                                                                         
v=0                                                                      
o=- 3815032750 3815032750 IN IP4 192.168.81.5
s=pjmedia     
b=AS:117                                                                                                                                          
t=0 0                                                                                                                                             
a=X-nat:1                                                                
m=audio 4010 UDP/TLS/RTP/SAVP 96 120                                     
c=IN IP4 192.168.81.5                                                    
b=TIAS:96000        
a=rtcp:4031 IN IP4 192.168.81.5
a=sendrecv                          
a=rtpmap:96 opus/48000/2                                                 
a=fmtp:96 useinbandfec=1                                                                                                                          
a=rtpmap:120 telephone-event/48000                                                                                                                
a=fmtp:120 0-16                                                                                                                                   
a=ssrc:1135627563 cname:297be1927ac33dc0                                                                                                          
a=ice-ufrag:01b20129                                                                                                                              
a=ice-pwd:4ab8de3f31f1da8b25358815                                                                                                                
a=candidate:Sc0a80107 1 UDP 1862270975 192.168.81.5 4010 typ srflx raddr 192.168.1.7 rport 4010                                                   
a=candidate:Hc0a80107 1 UDP 1694498815 192.168.1.7 4010 typ host                                                                                  
a=candidate:Sc0a80107 2 UDP 1862270974 192.168.81.5 4031 typ srflx raddr 192.168.1.7 rport 4031
a=candidate:Hc0a80107 2 UDP 1694498814 192.168.1.7 4031 typ host                                                                                  
a=rtcp-mux                                                                                                                                        
a=setup:actpass                                                                                                                                   
a=fingerprint:SHA-256 D1:DF:87:B0:28:D9:F0:33:13:8D:6A:60:FF:19:55:12:79:BE:A5:6D:FF:E6:DA:19:7E:51:52:EE:ED:FF:9D:F1                             
m=video 0 RTP/AVP 31                                                                                                                              
c=IN IP4 127.0.0.1                                                                                                                                

SIP/2.0 200 OK                                                                                                                                    
Via: SIP/2.0/TLS 10.13.20.1:40166;rport=40166;received=10.13.20.1;branch=z9hG4bKPj7X09kXL17bMarp1bKvl4pHOeG.R6fTiB;alias                          
Record-Route: <sip:example.com:5063;transport=tls;lr;r2=on>      
Call-ID: NUTfIMsRvuEVd0mhxrXqe9AwqylV753A                                                                                                         
From: <sip:mary@example.com>;tag=FyIJkATEQ0xXwh-yOLw7r63M0rvemI6V
To: <sip:alice@example.com>;tag=sgGS.Z2ZTSlYuaTbuqD4lTxArP.1dBik
CSeq: 13483 INVITE                                                                                                                                
Contact: <sip:alice@10.13.20.1:47762;transport=TLS;ob>;+sip.ice                                                        
Allow: PRACK, INVITE, ACK, BYE, CANCEL, UPDATE, INFO, SUBSCRIBE, NOTIFY, REFER, MESSAGE, OPTIONS                              
Supported: replaces, 100rel, timer, norefersub                                                                                                    
Session-Expires: 1800;refresher=uac                                                                                                               
Require: timer                                                                                                                                    
Content-Type: application/sdp                                            
Content-Length:   639                                                                                                                             
                                                                                                                                                  
v=0                                                                                                                                               
o=- 3815032751 3815032752 IN IP4 192.168.12.42                                                                                                    
s=pjmedia                                                                                                                                         
b=AS:117                                                                 
t=0 0                                                                                                                                             
a=X-nat:1                                                                
m=audio 17942 UDP/TLS/RTP/SAVP 96 120                                                                                                             
c=IN IP4 10.18.0.12                                                      
b=TIAS:96000                                                                                                                                      
b=RS:0          
b=RR:0                                                                                                                                            
a=ssrc:130475890 cname:7b1e8e2a5904f50e                                                                                                           
a=rtpmap:96 opus/48000/2                                                                                                                          
a=rtpmap:120 telephone-event/48000                                       
a=fmtp:96 useinbandfec=1
a=fmtp:120 0-16                                                          
a=sendrecv                                                               
a=rtcp:17942                       
a=rtcp-mux    
a=setup:active                                                                                                                                    
a=fingerprint:sha-1 32:AB:14:A0:A0:44:54:D3:12:F7:44:B6:57:48:46:CB:00:71:69:95
a=ice-ufrag:wo0zf6e0                                                     
a=ice-pwd:baciora3RusncKMd9sdDoGaDuG                                     
a=candidate:BE2FdFiRyWOsmhMl 1 UDP 2130706431 10.18.0.12 17942 typ host
m=video 0 RTP/AVP 31
c=IN IP4 10.18.0.12


UPDATE sip:alice@10.13.20.1:47762;transport=TLS;ob SIP/2.0
Via: SIP/2.0/TLS 10.13.20.1:40166;rport;branch=z9hG4bKPj49Pl-xO83zELp78e6ecJxr9vj69k6hRj;alias
Max-Forwards: 70
From: sip:mary@example.com;tag=FyIJkATEQ0xXwh-yOLw7r63M0rvemI6V
To: sip:alice@example.com;tag=sgGS.Z2ZTSlYuaTbuqD4lTxArP.1dBik
Contact: <sip:mary@10.13.20.1:40166;transport=TLS;extension=webrtc;ob>;+sip.ice
Call-ID: NUTfIMsRvuEVd0mhxrXqe9AwqylV753A
CSeq: 13484 UPDATE
Route: <sip:example.com:5063;transport=tls;lr;r2=on>
Supported: replaces, 100rel, timer, norefersub
Session-Expires: 1800;refresher=uac
Min-SE: 90    
Content-Type: application/sdp                                                                                                                     
Content-Length:   738
                                                                         
v=0                                                                      
o=- 3815032750 3815032751 IN IP4 192.168.81.5
s=pjmedia          
b=AS:117
t=0 0      
a=X-nat:1                                                                
m=audio 4010 UDP/TLS/RTP/SAVP 96 120                              
c=IN IP4 192.168.81.5                                                                                                                             
b=TIAS:96000                                                                                                                                      
b=RS:0                                                                   
b=RR:0                                                                                                                                            
a=rtpmap:96 opus/48000/2                                                                                                                          
a=fmtp:96 useinbandfec=1                                                                                                                          
a=rtpmap:120 telephone-event/48000                                                                                                                
a=fmtp:120 0-16                                                          
a=ssrc:1135627563 cname:297be1927ac33dc0                       
a=ice-ufrag:01b20129                                                                                                                              
a=ice-pwd:4ab8de3f31f1da8b25358815                                       
a=candidate:Sc0a80107 1 UDP 1862270975 192.168.81.5 4010 typ srflx raddr 192.168.1.7 rport 4010
a=remote-candidates:1 10.18.0.12 17942                                                                                                            
a=rtcp-mux                                                                                                                                        
a=setup:passive                                                                                                                                   
a=fingerprint:SHA-256 D1:DF:87:B0:28:D9:F0:33:13:8D:6A:60:FF:19:55:12:79:BE:A5:6D:FF:E6:DA:19:7E:51:52:EE:ED:FF:9D:F1
a=sendrecv                                                                                                                                        
m=video 0 RTP/AVP 31
c=IN IP4 127.0.0.1
a=inactive


SIP/2.0 200 OK
Via: SIP/2.0/TLS 10.13.20.1:40166;rport=40166;received=10.13.20.1;branch=z9hG4bKPj49Pl-xO83zELp78e6ecJxr9vj69k6hRj;alias
Call-ID: NUTfIMsRvuEVd0mhxrXqe9AwqylV753A
From: <sip:mary@example.com>;tag=FyIJkATEQ0xXwh-yOLw7r63M0rvemI6V
To: <sip:alice@example.com>;tag=sgGS.Z2ZTSlYuaTbuqD4lTxArP.1dBik
CSeq: 13484 UPDATE
Session-Expires: 1800;refresher=uac
Require: timer
Contact: <sip:alice@10.13.20.1:47762;transport=TLS;ob>;+sip.ice
Allow: PRACK, INVITE, ACK, BYE, CANCEL, UPDATE, INFO, SUBSCRIBE, NOTIFY, REFER, MESSAGE, OPTIONS
Supported: replaces, 100rel, timer, norefersub
Content-Type: application/sdp
Content-Length:   639

v=0
o=- 3815032751 3815032753 IN IP4 192.168.12.42
s=pjmedia
b=AS:117
t=0 0
a=X-nat:1
m=audio 17942 UDP/TLS/RTP/SAVP 96 120
c=IN IP4 10.18.0.12
b=TIAS:96000
b=RS:0
b=RR:0
a=ssrc:130475890 cname:7b1e8e2a5904f50e
a=rtpmap:96 opus/48000/2
a=rtpmap:120 telephone-event/48000
a=fmtp:96 useinbandfec=1
a=fmtp:120 0-16
a=sendrecv
a=rtcp:17942
a=rtcp-mux
a=setup:active
a=fingerprint:sha-1 32:AB:14:A0:A0:44:54:D3:12:F7:44:B6:57:48:46:CB:00:71:69:95
a=ice-ufrag:wo0zf6e0
a=ice-pwd:baciora3RusncKMd9sdDoGaDuG
a=candidate:BE2FdFiRyWOsmhMl 1 UDP 2130706431 10.18.0.12 17942 typ host
m=video 0 RTP/AVP 31
c=IN IP4 10.18.0.12
```

**Comment by space88man** [comment]
Closing this issue and reopening in a separate issue because the explanation here is bad.
